src/segments/nba.go

```diff
@@ -0,0 +1,425 @@
+	// Get the current date in America/New_York
+	t := time.Now().In(time.FixedZone("America/New_York", -5*60*60))
+
+	// Check to see if a game is scheduled while the numDaysToSearch is greater than 0
+	for numDaysToSearch > 0 {
+		// convert t into the format that the API expects "MM/YYYY/DD"
+		dateStr := fmt.Sprintf("%02d/%02d/%d", t.Month(), t.Day(), t.Year())
+		urlEndpoint := NBASchedURLPart1 + nbaSeason + "&Date=" + dateStr + NBASchedURLPart2
+		body, err := nba.env.HTTPRequest(urlEndpoint, nil, httpTimeout)
+		if err != nil {
+			return nil, err
+		}
+
+		var scheduleResponse *ScheduleResponse
+		err = json.Unmarshal(body, &scheduleResponse)
+		if err != nil {
+			return nil, err
+		}
+
+		// Check if we can find a game for the team
+		gameInfo, err := nba.findGameSchedulebyTeamTricode(scheduleResponse.ResultSets[1].CompleteGameList, teamName)
+		if err != nil {
+			// We didn't find a game for the team on this day, so we need to check the next day
+			t = t.AddDate(0, 0, 1)
+			numDaysToSearch--
+			continue
+		}
+
+		return &NBAData{
+			AwayTeam:       gameInfo.VtAbbreviation,
+			HomeTeam:       gameInfo.HtAbbreviation,
+			Time:           gameInfo.Time + " ET",
+			GameDate:       gameInfo.Date,
+			StartTimeUTC:   nba.parseTimetoUTC(gameInfo.Time, gameInfo.Date),
+			GameStatus:     Scheduled,
+			HomeScore:      0,
+			AwayScore:      0,
+			HomeTeamWins:   0,
+			HomeTeamLosses: 0,
+			AwayTeamWins:   0,
+			AwayTeamLosses: 0,
+		}, nil
+	}
+
+	return nil, errors.New("no scheduled game found for team within DaysOffset days")
+}
+
+// First try to get the data from the score endpoint, if that fails, try the schedule endpoint
+// The score endpoint usually goes live within 12 hours of a game starting
+func (nba *Nba) getAvailableGameData(teamName string, httpTimeout int) (*NBAData, error) {
+	// Get the info from the score endpoint
+	data, err := nba.retrieveScoreData(teamName, httpTimeout)
+	if err == nil {
+		return data, nil
+	}
+
+	// If the score endpoint doesn't have anything get data from the schedule endpoint
+	data, err = nba.retrieveScheduleData(teamName, httpTimeout)
+	if err == nil {
+		return data, nil
+	}
+
+	return nil, err
+}
+
+// Gets the data from the cache if it exists
+func (nba *Nba) getCacheValue(key string) (*NBAData, error) {
+	if val, found := nba.env.Cache().Get(key); found {
+		var nbaData *NBAData
+		err := json.Unmarshal([]byte(val), &nbaData)
+		if err != nil {
+			return nil, err
+		}
+		return nbaData, nil
+	}
+
+	return nil, errors.New("no data in cache")
+}
+
+// Gets the data from the cache for a scheduled game if it exists
+// Checks whether the game should have started and if so, removes the cache entry
+func (nba *Nba) getCachedScheduleValue(key string) (*NBAData, error) {
+	data, err := nba.getCacheValue(key)
+	if err != nil {
+		return nil, errors.New("no data in cache")
+	}
+
+	// check if the game was previously not found and we should wait to check again
```

Comment:
@JanDeDobbeleer - I'm actually experiencing a minor bug after testing a fake team tri-code "DSF". 

I think with this addition to the code that we return an error in the case that we have a game that was previously not found we are continuing to hit the endpoints. I can't leave a code review on my own PR GitHub says, so I'm tagging you in this comment instead.

Reply:
I'll check to see what we can do about that.

Reply:
This is solved now, it will only do the call again on the cache timeout and the context is logged.